platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/ResolvedArtifactSetResolver.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact;
+
+import org.gradle.api.internal.artifacts.ResolveArtifactsBuildOperationType;
+import org.gradle.api.internal.artifacts.configurations.ResolutionHost;
+import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.verification.DependencyVerificationOverride;
+import org.gradle.internal.operations.BuildOperationContext;
+import org.gradle.internal.operations.BuildOperationDescriptor;
+import org.gradle.internal.operations.BuildOperationExecutor;
+import org.gradle.internal.operations.RunnableBuildOperation;
+import org.gradle.internal.work.WorkerLeaseService;
+
+import javax.inject.Inject;
+
+/**
+ * Resolves a {@link ResolvedArtifactSet} in a build operation, visiting the results.
+ */
+public class ResolvedArtifactSetResolver {
+
+    private final WorkerLeaseService workerLeaseService;
+    private final BuildOperationExecutor buildOperationExecutor;
+    private final DependencyVerificationOverride dependencyVerificationOverride;
+
+    @Inject
+    public ResolvedArtifactSetResolver(
+        WorkerLeaseService workerLeaseService,
+        BuildOperationExecutor buildOperationExecutor,
+        DependencyVerificationOverride dependencyVerificationOverride
+    ) {
+        this.workerLeaseService = workerLeaseService;
+        this.buildOperationExecutor = buildOperationExecutor;
+        this.dependencyVerificationOverride = dependencyVerificationOverride;
+    }
+
+    public void visitInUnmanagedWorker(ResolvedArtifactSet artifacts, ArtifactVisitor visitor, ResolutionHost resolutionHost) {
```

Comment:
Perhaps this should be `visitInUnmanagedWorkerThread` to avoid confusion with the Workers API?